numba/core/typing/new_builtins.py

```diff
@@ -198,76 +655,107 @@ class BinOpMul(BinOp):
 
 @infer_global(operator.mod)
 class BinOpMod(ConcreteTemplate):
-    cases = list(integer_binop_cases)
+    cases = []
+
+    # Python cases
+    cases += [signature(op, op, op) for op in sorted(types.py_real_domain)]
```

Comment:
I don't think we really need these for the new type system. The ConcreteTemplate can all be removed and eventually be replaced by overload impl